Sources/XMTPiOS/Conversations.swift

```diff
@@ -99,33 +99,33 @@ public actor Conversations {
 	public func streamAllMessages() async throws -> AsyncThrowingStream<DecodedMessage, Error> {
 		return AsyncThrowingStream { continuation in
 			Task {
-				while true {
-					var topics: [String] = [
-						Topic.userInvite(client.address).description,
-						Topic.userIntro(client.address).description,
-					]
+				var topics: [String] = [
+					Topic.userInvite(client.address).description,
+					Topic.userIntro(client.address).description,
+				]
 
-					for conversation in try await list() {
-						topics.append(conversation.topic)
-					}
+				for conversation in try await list() {
+					topics.append(conversation.topic)
+				}
 
+				while(true) {
 					do {
-						for try await envelope in client.subscribe(topics: topics) {
+						for try await (envelope, subscription) in client.subscribe(topics: topics) {
```

Comment:
I tried a scenario where we returned the subscription independent of this for loop but the await function never completed. It may be my lack of knowledge of iOS threads/streams. But you can see that work in the previous commit history.

Reply:
@zombieobject I've got some iOS threading questions for you next week 🙏